app/controllers/questions_controller.rb

```diff
@@ -0,0 +1,63 @@
+class QuestionsController < ApplicationController
+
+  def index
+    @questions = Question.all
+  end
+
+  def new
+    if logged_in?
+      @question = Question.new
+    else
+      redirect_to root_path
+    end
+  end
+
+  def create
+    @question = current_user.questions.new(question_params)
+
+    if @question.save
+      tag_finder
+      redirect_to questions_path
+    else
+      @errors = @question.errors.full_messages.join(', ')
+      render :new
+    end
+  end
+
+  def show
+    @question = Question.find_by(id: params[:id])
+  end
+
+  def destroy
+  end
+
+  def update
+  end
+
+  def edit
+  end
+
+  private
+
+  def tag_finder
```

Comment:
Good thinking on splitting out this hairy work into a method. My only comment here is that this method modifies state of the controller but it's really hard to make a guess as to what it does when you see it somewhere like line 19 of this file. I'm a fan of naming it something more explicit like `load_tags` or `set_tags`.